Remotion/Web/ClientScript/Scripts/Utilities.ts

```diff
@@ -0,0 +1,347 @@
+    {
+      var majorVersion = parseInt (window.document.documentMode as any as string); // TODO RM-7631: window.document.documentMode does not need to be converted to a number
+      return majorVersion;
+    }
+    else if (navigator.appVersion.indexOf ('MSIE 7.0;') !== -1)
+    {
+      return 7;
+    }
+    else
+    {
+      return Number.NaN;
+    }
+  }
+}
+
+class PageUtility
+{
+  public static Instance: PageUtility;
+
+  private _resizeHandlers: PageUtility_ResizeHandlerItem[] = new Array();
+  private _resizeTimeoutID: Nullable<number> = null;
+  private _resizeTimeoutInMilliSeconds = 50;
+
+  constructor()
+  {
+    $(document).ready(function()
+    {
+      $(window).bind('resize', function() { PageUtility.Instance.PrepareExecuteResizeHandlers(); });
+    });
+  }
+
+  public RegisterResizeHandler (selector: string, handler: PageUtility_ResizeHandler): void
+  {
+    ArgumentUtility.CheckNotNullAndTypeIsString ('selector', selector);
+    ArgumentUtility.CheckNotNull ('handler', handler);
+
+    for (var i = 0; i < this._resizeHandlers.length; i++)
+    {
+      var item = this._resizeHandlers[i];
+      if (item.Selector == selector)
+      {
+        (item as any).handler = handler; // TODO 7632: PageUtility.RegisterResizeHandler assigns wrong property
+        return;
+      }
+    }
+    this._resizeHandlers[this._resizeHandlers.length] = new PageUtility_ResizeHandlerItem (selector, handler);
+  };
+
+  public PrepareExecuteResizeHandlers(): void
+  {
+    if (this._resizeTimeoutID != null)
+      window.clearTimeout (this._resizeTimeoutID);
+
+    this._resizeTimeoutID = window.setTimeout (function () { PageUtility.Instance.ExecuteResizeHandlers(); }, this._resizeTimeoutInMilliSeconds);
+  };
+
+  public ExecuteResizeHandlers(): void
+  {
+    var existingResizeHandlers = new Array();
+    for (var i = 0; i < this._resizeHandlers.length; i++)
+    {
+      var item = this._resizeHandlers[i];
+      var element = $ (item.Selector);
+      if (element != null && element.length > 0)
+      {
+        item.Handler (element);
+        existingResizeHandlers[existingResizeHandlers.length] = item;
+      }
+    }
+    this._resizeHandlers = existingResizeHandlers;
+  };
+
+  public IsInDom (element: HTMLElement): boolean
+  {
+    ArgumentUtility.CheckNotNull('element', element);
+
+    var html = window.document.body.parentNode;
+    let node: Nullable<Node & ParentNode> = element;
+    while (node)
+    {
+      if (node === html)
+          return true;
+      node = node.parentNode;
+    }
+    return false;
+  }
+}
+
+type PageUtility_ResizeHandler = (element: JQuery) => void;
+
+class PageUtility_ResizeHandlerItem
+{
+  public readonly Selector: string;
+  public Handler;
+
+  constructor(selector: string, handler: PageUtility_ResizeHandler)
+  {
+    this.Selector = selector;
+    this.Handler = handler;
+  }
+}
+
+PageUtility.Instance = new PageUtility();
+
+type WebServiceProxyInvokeFunction<TResult, TUserContext> = (
```

Comment:
Looks good, please add a Jira-ticket for updating the type definitions and add a comments in here referencing the ticket. in the Jira-Ticket we can then link to the Github issue in the 3rd party repo where we have reportet the issue.